javatests/arcs/android/host/AndroidSdkHandleManagerTest.kt

```diff
@@ -0,0 +1,262 @@
+import arcs.sdk.ReadableCollection
+import arcs.sdk.ReadableSingleton
+import arcs.sdk.WritableCollection
+import arcs.sdk.WritableSingleton
+import arcs.sdk.android.storage.service.DefaultConnectionFactory
+import arcs.sdk.android.storage.service.testutil.TestBindingDelegate
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.coroutines.experimental.suspendCoroutine
+
+typealias Person = TestParticleInternal1
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RunWith(AndroidJUnit4::class)
+class AndroidSdkHandleManagerTest {
+    private lateinit var app: Application
+
+    val entity1 = Person("Jason", 21.0, false)
+    val entity2 = Person("Jason", 22.0, true)
+
+    private val schema = Schema(
+        listOf(SchemaName("Person")),
+        SchemaFields(
+            singletons = mapOf(
+                "name" to FieldType.Text,
+                "age" to FieldType.Number,
+                "is_cool" to FieldType.Boolean
+            ),
+            collections = emptyMap()
+        ),
+        "1234acf"
+    )
+
+    private val singletonKey = ReferenceModeStorageKey(
+        backingKey = RamDiskStorageKey("single-back"),
+        storageKey = RamDiskStorageKey("single-ent")
+    )
+
+    private val setKey = ReferenceModeStorageKey(
+        backingKey = RamDiskStorageKey("set-back"),
+        storageKey = RamDiskStorageKey("set-ent")
+    )
+
+    @Before
+    fun setUp() {
+        app = ApplicationProvider.getApplicationContext()
+        app.setTheme(R.style.Theme_AppCompat);
+
+        // Initialize WorkManager for instrumentation tests.
+        WorkManagerTestInitHelper.initializeTestWorkManager(app)
+    }
+
+    fun handleManagerTest(block: suspend (SdkHandleManager) -> Unit) {
+        val scenario = ActivityScenario.launch(TestActivity::class.java)
+
+        scenario.moveToState(Lifecycle.State.STARTED)
+
+        scenario.onActivity { activity ->
+            runBlocking {
+                val hf = AndroidHandleManager(
+                    lifecycle = activity.lifecycle,
+                    context = activity,
+                    connectionFactory = DefaultConnectionFactory(activity, TestBindingDelegate(app))
+                )
+                block(SdkHandleManager(hf))
+            }
+        }
+
+        scenario.close()
+    }
+
+    private fun expectHandleException(
+        handleName: String,
+        block: () -> Unit
+    ) {
+        try {
+            block()
+        } catch (e: Exception) {
+            assertThat(e).isInstanceOf(NoSuchElementException::class.java)
+            assertThat(e.message).isEqualTo(
+                "Handle ${handleName} not initialized in TestParticle"
+            )
+        }
+    }
+
+    @Test
+    fun testCreateSingletonHandle() = runBlockingTest {
```

Comment:
nit: `test` prefix usually not used for test method names when using JUnit that has `@Test` annotation, since it's implicit, same rest of PR.